linera-faucet/server/src/tests.rs

```diff
@@ -88,114 +89,187 @@ async fn test_faucet_rate_limiting() {
-
-    let faucet_storage = Arc::new(Mutex::new(FaucetStorage::default()));
-    let temp_dir = tempdir().unwrap();
-    let storage_path = temp_dir.path().join("test_faucet_persistence.json");
+    }));
 
-    // Set clock past any time limits to avoid balance checking issues
-    clock.set(Timestamp::from(u64::MAX));
+    let faucet_storage = Arc::new(Mutex::new(super::FaucetStorage::default()));
+    let pending_requests = Arc::new(Mutex::new(VecDeque::new()));
+    let request_notifier = Arc::new(Notify::new());
 
-    let root = MutationRoot {
+    // Create batch processor with initial batch size of 3 and disabled rate limiting
+    let initial_batch_size = 3;
+    let config = super::BatchProcessorConfig {
         chain_id,
-        context: context.clone(),
         amount: Amount::from_tokens(1),
-        end_timestamp: Timestamp::from(0), // Already passed, so no time limits
-        start_timestamp: Timestamp::from(0),
-        start_balance: Amount::from_tokens(10),
-        faucet_storage,
+        start_balance: Amount::from_tokens(100),
+        start_timestamp: Timestamp::from(1000), // start > end disables rate limiting
+        end_timestamp: Timestamp::from(999),
         storage_path: storage_path.clone(),
+        max_batch_size: initial_batch_size,
     };
 
-    let test_owner = AccountPublicKey::test_key(0).into();
-
-    // First claim should succeed and create a new chain
-    let first_description = root.do_claim(test_owner).await.unwrap();
+    let mut batch_processor = super::BatchProcessor::new(
+        config,
+        Arc::clone(&context),
+        Arc::clone(&faucet_storage),
+        Arc::clone(&pending_requests),
+        Arc::clone(&request_notifier),
+    );
 
-    // Second claim with the same owner should return the same chain
-    let second_result = root.do_claim(test_owner).await;
-    assert!(second_result.is_ok());
-    let second_description = second_result.unwrap();
+    // Create 3 different owners for batch processing
+    let owners = [
+        AccountOwner::from_str("0x1111111111111111111111111111111111111111").unwrap(),
+        AccountOwner::from_str("0x2222222222222222222222222222222222222222").unwrap(),
+        AccountOwner::from_str("0x3333333333333333333333333333333333333333").unwrap(),
+    ];
 
-    // Should be the same chain
-    assert_eq!(first_description.id(), second_description.id());
-}
+    // Create and queue 3 pending requests
+    {
+        let mut pending_requests_guard = pending_requests.lock().await;
+        for owner in owners {
+            let (tx, _rx) = oneshot::channel();
+            pending_requests_guard.push_back(super::PendingRequest {
+                owner,
+                responder: tx,
+            });
+        }
+    }
 
-#[test]
-fn test_multiply() {
-    use super::multiply;
+    // Execute the batch - this triggers BlockTooLarge error
+    batch_processor
+        .process_batch()
+        .await
+        .expect("Batch processing should succeed");
```

Comment:
What is the reason this batch is too large? `maximum_block_size`?

Reply:
Yes. I couldn't come up with a better way to cause one of those error types.